opencog/atoms/execution/EvaluationLink.cc

```diff
@@ -475,7 +475,11 @@ TruthValuePtr EvaluationLink::do_eval_scratch(AtomSpace* as,
 	{
 		return do_eval_scratch(as, DefineLink::get_definition(evelnk), scratch);
 	}
-	else if (INHERITANCE_LINK == t or IMPLICATION_LINK == t)
+	else if (// Links that evaluate to themselves
+		INHERITANCE_LINK == t or
+		IMPLICATION_LINK == t or
+		EXECUTION_LINK == t
```

Comment:
This is incorrect. Execution links are executable. They are not evaluatable.  If you want evaluation, ue an EvaluationLink. If you want execution, use an ExecutionLink!

Reply:
FWIW, I am maybe over-stating my case.  Now that we have generic values, it is perhaps the right time to merge ExecutionLink and EvaluationLink into one.    Note that both TV's and Atoms are special cases of values.

Specifically, a ValuationLink could be any link that returns a value, EvaluationLink could be any ValuationLlink that returns a truth value, and an ExecutionLink is any kind of ValuationLink that returns an atom.

Reply:
@linas an `ExecutionLink` doesn't execute anything, it keeps a record what has been executed, or is expected to be executed, as explained here http://wiki.opencog.org/w/ExecutionLink

I agree that the terminology can be confusing, I'm used to it, but maybe others would prefer something like
`ExecutionOutputLink` -> `ApplicationLink`
`ExecutionLink` -> `ApplicationRecordLink`